conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,6202 @@
+                                // @output false
+    subtypeIntFloat(0X3eA3P34); // @output true
+                                // @output false
+    subtypeIntFloat(0x3eA3p+34); // @output true
+                                // @output false
+    subtypeIntFloat(0x3eA3P-34); // @output true
+                                // @output false
+    subtypeIntFloat(0X3eA3p+34); // @output true
+                                // @output false
+    subtypeIntFloat(0X3eA3P-34); // @output true
+                                // @output false
+    subtypeIntFloat(0x3eA3.4Eb2); // @output true
+                                    // @output false
+    subtypeIntFloat(0x.4Eb2); // @output true
+                                // @output false
+    subtypeIntFloat(0X3eA3.4Eb2); // @output true
+                                    // @output false
+    subtypeIntFloat(0X.4Eb2); // @output true
+                                // @output false
+    subtypeIntFloat(0x3eA3.4Eb2p45); // @output true
+                                    // @output false
+    subtypeIntFloat(0x3eA3.4Eb2P45); // @output true
+                                    // @output false
+    subtypeIntFloat(0x.4Eb2p45); // @output true
+                                // @output false
+    subtypeIntFloat(0x.4Eb2P45); // @output true
+                                // @output false
+    subtypeIntFloat(0X3eA3.4Eb2p45); // @output true
+                                // @output false
+    subtypeIntFloat(0X3eA3.4Eb2P45); // @output true
+                            // @output false
+    subtypeIntFloat(0X.4Eb2p45); // @output true
+                            // @output false
+    subtypeIntFloat(0X.4Eb2P45); // @output true
+                            // @output false
+    subtypeIntFloat(0x3eA3.4Eb2p+45); // @output true
+                                // @output false
+    subtypeIntFloat(0x3eA3.4Eb2P-45); // @output true
+                                    // @output false
+    subtypeIntFloat(0x.4Eb2p+45); // @output true
+                                // @output false
+    subtypeIntFloat(0x.4Eb2P-45); // @output true
+                                // @output false
+    subtypeIntFloat(0X3eA3.4Eb2p+45); // @output true
+                                    // @output false
+    subtypeIntFloat(0X3eA3.4Eb2P-45); // @output true
+                                // @output false
+    subtypeIntFloat(0X.4Eb2p+45); // @output true
+                                // @output false
+    subtypeIntFloat(0X.4Eb2P-45); // @output true
+                                // @output false
+    subtypeIntFloat(0x12aB); // @output false
+                                // @output true
+}
+
+function subtypeIntFloat(int|float x) {
+    io:println(x is float);
+    io:println(x is int);
+}
+
+Test-Case: output
+Description: Test the basic type when the applicable contextually expected type is a subtype of float, having a union with nil and float types.
```

Comment:
OK to keep, but why were these tests added? What exactly was the special scenario we wanted to test?